mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -800,11 +808,25 @@ impl MakerSwap {
             })
             .compat();
 
+        let reward_amount = if self.r().watcher_reward {
+            let reward = match watcher_reward_amount(&self.maker_coin, &self.taker_coin).await {
+                Ok(reward) => reward,
+                Err(err) => {
+                    return Ok((Some(MakerSwapCommand::Finish), vec![
+                        MakerSwapEvent::MakerPaymentTransactionFailed(err.into_inner().to_string().into()),
+                    ]))
+                },
```

Comment:
Maybe we can retry a few times in case there is a problem in the RPC node before failing the whole swap similar to how it's done in fee validation https://github.com/KomodoPlatform/atomicDEX-API/blob/0d7e859b7d372397575e09663a412e30e2ed2ec5/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L765